packages/uicore/src/components/ModalDialog/ModalDialog.tsx

```diff
@@ -65,25 +64,58 @@ export const ModalDialog: FC<ModalDialogProps> = ({
   ...dialogProps
 }) => {
   const bodyRef = useRef<HTMLDivElement>(null)
-
-  const manageScrollIndicators = throttle(() => {
-    const el = bodyRef.current
-
-    if (el) {
-      el.style.setProperty('--showScrollTop', el.scrollTop > 0 ? '1' : '0')
-      el.style.setProperty('--showScrollBottom', el.scrollTop < el.scrollHeight - el.clientHeight ? '1' : '0')
-    }
-  }, 100)
+  const bodyTopEdgeRef = useRef<HTMLDivElement>(null)
+  const bodyBottomEdgeRef = useRef<HTMLDivElement>(null)
+  const bodyObserverRef = useRef<IntersectionObserver>()
+
+  const [scrollShadows, setScrollShadows] = useState({ top: false, bottom: false })
+
+  const bodyShadowClass = useMemo(() => {
+    const { top, bottom } = scrollShadows
+    if (!top && !bottom) return ''
+    if (top && bottom) return css.shadowTopAndBottom
+    return top ? css.shadowTop : css.shadowBottom
+  }, [scrollShadows])
+
+  const observeEdge = useCallback((element: HTMLDivElement | null | undefined, observer: IntersectionObserver) => {
```

Comment:
No harm in it, but this doesn't really need to be a callback, it could just be a standard function.

Reply:
Used `useCallback` here as this function is added to the dependencies array of other function (`initScrollShadows`) that's using `useCallback`.

Reply:
It's fine either way, but if you were to alter this to a `function` outside of the scope of the component, you wouldn't need to wrap it in a callback as it would never be recreated